pretty_gpx/gpx/gpx_track.py

```diff
@@ -46,6 +51,32 @@ def load(gpx_path: str | bytes) -> tuple['GpxTrack', float, float]:
                     gpx_track.list_lon.append(point.longitude)
                     gpx_track.list_lat.append(point.latitude)
                     gpx_track.list_ele.append(point.elevation)
+                    if idx == 0:
+                        d = all_segment_cumul_d
+                        previous_cum_d = all_segment_cumul_d
+                        h_diff = all_segment_cumul_ele
+                        previous_cum_ele = all_segment_cumul_ele
+                    else:
+                        d = gpxpy.geo.distance( point_prev.latitude,
+                                                point_prev.longitude,
+                                                point_prev.elevation,
+                                                point.latitude,
+                                                point.longitude,
+                                                point.elevation)*1e-3
+                        h_diff = point.elevation - point_prev.elevation
+                        if h_diff < 0:
+                            h_diff = 0.0
+                    
+                    gpx_track.list_cumul_d.append(previous_cum_d+d)
+                    gpx_track.list_cumul_ele.append(previous_cum_ele+h_diff)
+
+                    previous_cum_d += d
+                    previous_cum_ele += h_diff
+
+                    point_prev = point
+                all_segment_cumul_d += previous_cum_d
+                all_segment_cumul_ele += previous_cum_ele
+
```

Comment:
maybe you can add the assert at the end of this method to compare the cum sums against `gpx.length_3d()*1e-3` and `gpx.get_uphill_downhill().uphill` ?